lib/modes/search.ex

```diff
@@ -1,9 +1,63 @@
 defmodule SonicClient.Modes.Search do
+  @moduledoc """
+  Module to search data on sonic.
+  """
   alias SonicClient
   alias SonicClient.TcpConnection
 
-  def query(conn, collection, bucket, terms) do
-    command = ~s(QUERY #{collection} #{bucket} "#{terms}")
+  @default_limit 15
+
+  @doc """
+  Query sonic database
+
+  ## Examples
+
+  iex> {:ok, conn} = SonicClient.start(127.0.0.1, 1491, "search", "secret")
+
+  iex> SonicClient.Modes.Search.query("my_collection", "bucket", "test")
+  {:ok, ["object1", "object2"]}
+
+  iex> SonicClient.Modes.Search.query("my_collection", "bucket", "test", limit: 1)
+  {:ok, ["object1"]}
+
+  iex> SonicClient.Modes.Search.query("my_collection", "bucket", "test", limit: 1, offset: 1)
+  {:ok, ["object2"]}
+
+  """
+  def query(conn, collection, bucket, terms, opts \\ [limit: 10, offset: 0]) do
```

Comment:
Yeah, maybe we need to agree on a common interface for the functions, i.e. if we want to define multiple signatures for the same name or go with default values (may not work in `PUSH`). Good for now, though!

Reply:
Yeah we definitely need that. I don't have any strong opinion which one to use.